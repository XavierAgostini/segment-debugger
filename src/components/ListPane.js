import React from 'react'
import ListControlBar from './ListControlBar'
import EventList from './EventList'
import './ListPane.module.css'

export default class ListPane extends React.Component {
  render() {
    return (
      <div className="listPane">
        <ListControlBar 
          handlePauseChange={this.props.handlePauseChange}
          handleSearchFilter={this.props.handleSearchFilter}
        />
        <EventList
          events={this.props.events} 
          handleEventSelected={this.props.handleEventSelected} 
          selectedEvent={this.props.selectedEvent}
        />
      </div>
    )
  }
}