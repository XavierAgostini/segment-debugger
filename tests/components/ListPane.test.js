import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListPane from '../../src/components/ListPane'

// w