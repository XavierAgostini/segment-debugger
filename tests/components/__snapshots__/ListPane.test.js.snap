// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListPane/> should render ListPane as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListPane
    handleEventSelected={[MockFunction]}
    handlePauseChange={[MockFunction]}
    handleSearchFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ListControlBar
          handlePauseChange={[MockFunction]}
          handleSearchFilter={[MockFunction]}
        />,
        <EventList
          handleEventSelected={[MockFunction]}
        />,
      ],
      "className": "listPane",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handlePauseChange": [MockFunction],
          "handleSearchFilter": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "events": undefined,
          "handleEventSelected": [MockFunction],
          "selectedEvent": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ListControlBar
            handlePauseChange={[MockFunction]}
            handleSearchFilter={[MockFunction]}
          />,
          <EventList
            handleEventSelected={[MockFunction]}
          />,
        ],
        "className": "listPane",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handlePauseChange": [MockFunction],
            "handleSearchFilter": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "events": undefined,
            "handleEventSelected": [MockFunction],
            "selectedEvent": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
